Vertexvis/vertex-web-sdk
# [SDK-797] Support CI releases via Lerna

## What

Updates our CI/CD pipeline to use `lerna publish from-package` internally. This will automatically check if a version exists in NPM and if not, publish the package.

Made some workflow changes to the pipeline as well. There's now two workflows:

* `branch.yml`: Is run on any git branch. It contains 3 jobs: `build`, `test` and `lint`. Lint and build run in parallel, so you'll get notified sooner if that stage fails.
* `master.yml`: Is run on any merge to master. It contains 3 jobs: `build`, `detect-publish`, `publish`. Publish will only run if a version change is detected in the `lerna.json` file. If a version change is detected, we'll publish packages to NPM and create a GH release.

Also added dependency caching to reduce the time for builds.

There's some setup duplication among the scripts. Plan is to pull these out into a GH action so we can share across jobs / repos.

## Ticket

https://vertexvis.atlassian.net/browse/SDK-797

## Test Plan

1. Merge
2. Create a relaese
2. Verify that packages are released

## Areas of Possible Regression

N/A

## Out of scope changes made in PR

N/A

## Dependencies

N/A
